src/cattr/gen.py

```diff
@@ -92,9 +95,31 @@ def make_dict_unstructure_fn(cl, converter, omit_if_default=False, **kwargs):
 
 def make_dict_structure_fn(cl: Type, converter, **kwargs):
     """Generate a specialized dict structuring function for an attrs class."""
+
+    mapping = kwargs.get("mapping", None)
+    if is_generic(cl):
+        base = get_origin(cl)
+        mapping = converter._generate_mapping(cl, mapping)
+        cl = base
+
+    for base in getattr(cl, "__orig_bases__", ()):
+        if is_generic(base) and not str(base).startswith("typing.Generic"):
+            mapping = converter._generate_mapping(base, mapping)
+            break
+
+    if isinstance(cl, TypeVar):
+        cl = getattr(mapping, cl.__name__, cl)
+
     cl_name = cl.__name__
     fn_name = "structure_" + cl_name
-    globs = {"__c_s": converter.structure, "__cl": cl}
+    for p in getattr(cl, "__parameters__", ()):
```

Comment:
This block is to come up with a pretty name for the function? Can leave a comment.